algorithms/xc20p.js

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import crypto from '../crypto.js';
+import {XChaCha20Poly1305, KEY_LENGTH} from '@stablelib/xchacha20poly1305';
+
+export const JWE_ENC = 'XC20P';
+
+/**
+ * Generates a content encryption key (CEK). The 256-bit key is intended to be
+ * used as a XChaCha20Poly1305 (draft-irtf-cfrg-xchacha-01) key.
+ *
+ * @returns {Promise<Uint8Array>} Resolves to the generated key.
+ */
+export async function generateKey() {
+  // generate content encryption key
+  return crypto.getRandomValues(new Uint8Array(KEY_LENGTH));
+}
+
+/**
+ * Encrypts some data. The data will be encrypted using the given
+ * 256-bit XChaCha20Poly1305 (draft-irtf-cfrg-xchacha-01) content encryption
+ * key (CEK).
+ *
+ * @param {object} options - The options to use.
+ * @param {Uint8Array} options.data - The data to encrypt.
+ * @param {Uint8Array} options.additionalData - Optional additional
```

Comment:
```suggestion
 * @param {Uint8Array} [options.additionalData] - Optional additional
```
Maybe having 'Optional' in the description with the proper jsdoc annotation is not required.